src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -465,6 +466,7 @@ class AmazonWebServicesProvider(schema.Base):
     kubernetes_version: str
     availability_zones: Optional[List[str]]
     node_groups: Dict[str, AWSNodeGroup] = DEFAULT_AWS_NODE_GROUPS
+    eks_endpoint_access: str = "public"
```

Comment:
I think having an Enum here (https://docs.pydantic.dev/2.2/usage/types/enums/) rather than a string would make sense. That way you don't need to write your own validator.